spec/command_spec.rb

```diff
@@ -77,13 +83,17 @@ module Fauxpaas
               restart: OpenStruct.new(success?: true)),
             source: double(:source, latest: double(:latest)),
             log_release: true,
-            signature: double(:signature),
+            signature: signature,
             releases: ["one", "two", "three"]
           )
         end
         before(:each) do
-          allow(Release).to receive(:new)
-            .with(signature: instance.signature,fs: anything).and_return(release)
+          allow(DeployConfig).to receive(:from_ref).with(signature.deploy, Fauxpaas.ref_repo)
+            .and_return(deploy_config)
+          allow(Artifact).to receive(:new).with(signature: signature, ref_repo: Fauxpaas.ref_repo)
```

Comment:
I was trying to get out of the habit of replacing `new` on classes in tests in preference to injecting a stubbed class - replacing the class method seems like a fairly heavy-handed thing that could potentially have unexpected consequences elsewhere in the app. In this particular case I think it's OK -- we've done it to all the collaborators here, and that keeps it as a true unit test and ensures the stubbed `new` doesn't "escape" outward -- but in general, I'd prefer to try not to make a habit of this.